### PR TITLE
[MIG] event_project: Migration to 10.0

### PR DESCRIPTION
**Pull request by chienandalu**
Plan events through projects
============================

This module allows you to assign a project to an event. This project will
inherit its event's name an its due date will be set as the event's begining
date.

Depends on:

- [x] project_recalculate: https://github.com/OCA/project/pull/291

cc @Tecnativa

### CODE REVIEW

File: event_project/README.rst [hunk 1]
```diff
@@ -1,66 +1,39 @@
...
 =====
 
 To use this module, you need to:
 
-* First step, create a project template.
-* Second step, indicate in to task of template the previous days to begin task.
-* And Last step, create a new event and select project template.
-
-It will then create automatically a new project related with the event.
+* Create a project or duplicate an existing one that fits your needs.
```
Comment (pedrobaeza) [inline event_project/README.rst]: Use numbered bullets with `#.`

File: event_project/README.rst [hunk 2]
```diff
@@ -1,66 +1,39 @@
...
+* Go to an event or create a brand new one and assign the previous project.
+* The project will inherit its
```
Comment (pedrobaeza) [inline event_project/README.rst]: This sentence is cut.

File: event_project/README.rst [hunk 3]
```diff
@@ -1,66 +1,39 @@
...
 Plan events through projects
 ============================
 
-This module allows you to assign a project template to an event, creating the
-corresponding project for the event that ends on the event start date, and
-plan the project tasks backwards according a new field "Event anticipation days".
-
-If the event date is changed, then all the pending tasks are retro-planned
-again according the new date.
+This module allows you to assign a project to an event. This project will
```
Comment (pedrobaeza) [inline event_project/README.rst]: The idea is to make a copy of the project, not to use it.

File: event_project/__manifest__.py [hunk 4]
```diff
@@ -1,26 +1,28 @@
...
+# Copyright 2015 Antiun - Endika Iglesias
+# Copyright 2015 Tecnativa - Antonio Espinosa
+# Copyright 2016 Tecnativa - Pedro M. Baeza
+# Copyright 2017 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 {
     "name": "Event project",
-    "version": "8.0.1.1.1",
+    "version": "10.0.1.0.0",
     "author": "Antiun Ingeniería S.L., "
```
Comment (pedrobaeza) [inline event_project/__manifest__.py]: Remove Antiun also

File: event_project/wizard/project_template_wizard.py [hunk 5]
```diff
@@ -1,32 +0,0 @@
-# -*- coding: utf-8 -*-
-# See README.rst file on addon root folder for license details
-
-from openerp import models, fields, exceptions, api
-from openerp.tools.translate import _
-
-
-class ProjectTemplateWizard(models.TransientModel):
```
Comment (pedrobaeza) [inline event_project/wizard/project_template_wizard.py]: Why do you remove this wizard? Just drop the domain for "template" projects, but don't remove it.

File: event_project/views/event_event_view.xml [hunk 6]
```diff
@@ -18,39 +17,20 @@
...
-                class="oe_stat_button oe_inline"
-                icon="fa-tasks">
+                    type="action"
+                    class="oe_stat_button oe_inline"
+                    icon="fa-tasks"
+                    attrs="{'invisible': [('count_tasks','&lt;','1')]}">
                 <field name="count_tasks" widget="statinfo" string="Tasks"/>
             </button>
         </div>
-        <page string="Event Details" position="after">
```
Comment (pedrobaeza) [inline event_project/views/event_event_view.xml]: We need to add a module _event_project_timesheet_ that restores back the equivalent of this page.

**Comment by pedrobaeza** [comment]
Please rebase and make sure OCA/project is on oca_dependencies.txt

**Comment by chienandalu** [comment]
@pedrobaeza All green now

**Review by pedrobaeza — APPROVED** [review]

File: event_project/README.rst [hunk 7]
```diff
@@ -1,23 +1,39 @@
...
 
 To use this module, you need to:
 
-* First step, create a project template.
-* Second step, indicate in to task of template the previous days to begin task.
-* And Last step, create a new event and select project template.
-It will then create automatically a new project related with the event.
+* Create a project or duplicate an existing one that fits your needs.
+* Go to an event or create a brand new one and assign the previous project.
+* The project will inherit its
```
Comment (pedrobaeza) [inline event_project/README.rst]: This text is still half

File: event_project/README.rst [hunk 8]
```diff
@@ -1,23 +1,39 @@
...
 Usage
 =====
 
 To use this module, you need to:
 
-* First step, create a project template.
-* Second step, indicate in to task of template the previous days to begin task.
-* And Last step, create a new event and select project template.
-It will then create automatically a new project related with the event.
+* Create a project or duplicate an existing one that fits your needs.
```
Comment (pedrobaeza) [inline event_project/README.rst]: Don't remove the feature of making a project copy

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
@chienandalu please review my last comments

**Comment by chienandalu** [comment]
@pedrobaeza Changes done

**Comment by chienandalu** [comment]
@pedrobaeza I'm triying to recover the project copy feature but I'm having issues on event write().
Everything goes fine until here (https://github.com/Tecnativa/event/blob/17a4ace53f5e5681e0df3ec50d8eeaa9470840b6/event_project/models/event_event.py#L73). There is a copy of the project and it's linked to the event, but once it goes out from write(), event.project_id resets to its previous value :confused: 

Any ideas?

File: event_project/models/event_event.py [hunk 9]
```diff
@@ -1,72 +1,87 @@
...
 
-    project_template = fields.Many2one(
-        comodel_name='project.project', string='Template project',
-        domain="[('state', '=', 'template')]")
-    project = fields.Many2one(
+    project_id = fields.Many2one(
         comodel_name='project.project', string='Related project',
-        readonly=True)
+        help="Project end date will be updated with event start date.")
+    task_ids = fields.One2many(
```
Comment (pedrobaeza) [inline event_project/models/event_event.py]: A related `One2many` without `readonly`??!! I don't think so... This for sure is causing troubles.

**Comment by pedrobaeza** [comment]
The related one2many is still there. You can't make a related one2many AFAIK.

**Comment by pedrobaeza** [comment]
Check conflicting files also.

**Comment by chienandalu** [comment]
@pedrobaeza Conflicts resolved but still having the same issue:

Here everything is set ok: https://github.com/Tecnativa/event/blob/17a4ace53f5e5681e0df3ec50d8eeaa9470840b6/event_project/models/event_event.py#L73
Here Odoo changes it back: https://github.com/odoo/odoo/blob/10.0/odoo/fields.py#L909

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
In this case, the problem is not real, but you have to force a refresh of the record. Check d38d90dd3e8b24bf8bf3d6a6f101d715fae98064 for the solution (call `refresh`), and other patches to avoid multi-record problems (take always into account that any button/method with `@api.multi` can be called with multiple records).

**Comment by pedrobaeza** [comment]
@luismontalba please review

**Comment by chienandalu** [comment]
Thanks @pedrobaeza :slightly_smiling_face:

**Review by luismontalba — APPROVED** [review]
Tested on runbot

**Pull request merged**